skcosmo/_selection.py

```diff
@@ -570,14 +571,16 @@ def _compute_pi(self, X, y=None):
     def _update_post_selection(self, X, y, last_selected):
         """
         Saves the most recently selected feature, increments the feature counter,
-        and, if the CUR is iterative, orthogonalizes the remaining features by
+        and, if the CUR is iterative (recompute_every>0), orthogonalizes the remaining features by
         the most recently selected.
         """
         super()._update_post_selection(X, y, last_selected)
 
-        if self.iterative:
+        if self.recompute_every != 0:
             self._orthogonalize(last_selected)
-            self.pi_ = self._compute_pi(self.X_current_)
```

Comment:
Does it really make sense to orthogonalize t every step?  Maybe it should be done every `recompute_every` steps, too?

Reply:
The orthogonalizers work on column vectors, so it is best to do step by step -- the `_compute_pi` is the real time-sink, so that should fix the time complexity issue.